### PR TITLE
Rename promicro_rp2040 converter to sparkfun_pm2040

### PR DESCRIPTION
**Pull request by daskygit**
## Description

My assumption is that this can be merged next cycle and the intention of renaming can be called out this cycle. Or should backwards compatibility be maintained for a couple of cycles?

I believe this naming is causing quite a few support questions due to the missing vbus detection on the sparkfun version vs the clones.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
My suggestion would be to do this cycle,
* rename platforms/chibios/converters/promicro_to_promicro_rp2040
* add alias for old name
* current docs changes from this PR

That way we can at least start pointing people to the new name.

**Comment by tzarc** [comment]
Can you add a changelog file as well so I don’t forget to include it?

**Review by zvecr — APPROVED** [review]

File: docs/ChangeLog/PR24192.md
Comment (zvecr) [inline docs/ChangeLog/PR24192.md:1]: Missed the location on this, can it be moved to `docs/ChangeLog/20240825/PR24192.md`

Comment (daskygit) [reply docs/ChangeLog/PR24192.md:1]: done

**Review by drashna — APPROVED** [review]

**Pull request merged**